xapi/design/fcoe-nics.md

```diff
@@ -2,53 +2,50 @@
 title: FCoE capable NICs
 layout: default
 design_doc: true
-revision: 2
+revision: 3
 status: proposed
 design_review: 120
 ---
 
 It has been possible to identify the NICs of a Host which can support FCoE.
-This property can be listed in PIF object under capabilities field with key `FCoE`.
+This property can be listed in PIF object under capabilities field.
 
 Introduction
 ------------
 
 * FCoE supported on a NIC is a hardware property. With the help of dcbtool, we can identify which NIC support FCoE.
-* The new field capabilites will be `(string -> string) map` in PIF object. For FCoE capability in PIF capabilties will have field set to `"FCoE" -> "Unsupported"`.
-* `Supported` string will state NIC supports FCoE. `Unsupported` string will state NIC does not supports FCoE.
-* Capabilties `(key, value)` field will be ReadOnly, This field cannot be modified by user.
+* The new field capabilities will be `Set(String)` in PIF object. For FCoE capable NIC will have string "fcoe" in PIF capabilities field.
+* `capabilities` field will be ReadOnly, This field cannot be modified by user.
 
 PIF Object
 -------
 
 New field:
-* Field `PIF.capabilties` will be type `(string -> string) map`.
-* Default value of PIF capabilties will have field set to `"FCoE" -> "Unsupported"`.
+* Field `PIF.capabilities` will be type `Set(string)`.
+* Default value in PIF capabilities will have empty string "".
 
 Xapi Changes
 ------
 
-* Set the field capabilities `FCoE` key to `Unsupported` or `Supported` depending on output of xcp-networkd call `is_fcoe_supported`.
-* Field capabilities `FCoE` can be set during `introduce_internal` when the PIF is created.
-* Field capabilties `FCoE` can be updated during `refresh_all` during PIF scan.
-* The above field will be set everytime when xapi-restart.
-
-XCP-Networkd Changes
-------
-
 New function:
-* Boolean `bool is_fcoe_supported (string)`
-  Argument: the device_name for the PIF.
-* This function returns `true` or `false` depending on dcbtool output for the device provided.
+* String `string get_fcoe_capability (string)`
```

Comment:
I'd prefer a `get_capabilities` function in xcp-networkd's interface. This would internally figure out whether fcoe is supported and return `["fcoe"]` or `[]`.